src/infi/systray/win32_adapter.py

```diff
@@ -80,11 +82,25 @@
     LRESULT = ctypes.c_longlong
 HANDLE = ctypes.c_void_p
 
-def convert_to_ascii(s):
+# On Win 2000 and later, tray hover text can be 128 characters, otherwise 64
+# https://msdn.microsoft.com/en-us/library/windows/desktop/bb773352%28v=vs.85%29.aspx
+# https://msdn.microsoft.com/en-us/library/ms724834%28v=vs.85%29.aspx
+if sys.getwindowsversion().major >= 5:
+    SZTIP_MAX_LENGTH = 128 
+else:
+    SZTIP_MAX_LENGTH = 64

```

Comment:
This is quite surprising! Looking at shellapi.h, I see that there are big differences in the NOTIFYICONDATA struct between several versions: pre-2000, pre-XP, pre-Vista and post-Vista. The struct with `szTip[64]` doesn't contain all other members after `szTip` so the struct defined here should actually not match both versions (szTip[64] means we're using the pre-2000 version, but the other members mean we're using the latest, post-Vista version)
I'm not sure how the code here works, then :) but it does.
Do you think we should match the struct to all possible versions? I'm not sure how that would make the code look.